crates/router/src/connector/braintree/braintree_graphql_transformers.rs

```diff
@@ -0,0 +1,988 @@
+// pub struct BraintreeAuthType {
+//     pub(super) auth_header: String,
+//     pub(super) merchant_id: Secret<String>,
+// }
+
+// impl TryFrom<&types::ConnectorAuthType> for BraintreeAuthType {
+//     type Error = error_stack::Report<errors::ConnectorError>;
+//     fn try_from(item: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
+//         if let types::ConnectorAuthType::SignatureKey {
+//             api_key: public_key,
+//             key1: merchant_id,
+//             api_secret: private_key,
+//         } = item
+//         {
+//             let auth_key = format!("{}:{}", public_key.peek(), private_key.peek());
+//             let auth_header = format!("Basic {}", consts::BASE64_ENGINE.encode(auth_key));
+//             Ok(Self {
+//                 auth_header,
+//                 merchant_id: merchant_id.to_owned(),
+//             })
+//         } else {
+//             Err(errors::ConnectorError::FailedToObtainAuthType)?
+//         }
+//     }
+// }
+
+#[derive(Debug, Default, Clone, Deserialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum BraintreePaymentStatus {
+    Authorized,
+    Authorizing,
+    AuthorizedExpired,
+    Failed,
+    ProcessorDeclined,
+    GatewayRejected,
+    Voided,
+    #[default]
+    Settling,
+    Settled,
+    SettlementPending,
+    SettlementDeclined,
+    SettlementConfirmed,
+    SubmittedForSettlement,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct ErrorDetails {
+    pub message: String,
+    pub extensions: Option<AdditionalErrorDetails>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AdditionalErrorDetails {
+    pub legacy_code: Option<String>,
+}
+
+impl From<BraintreePaymentStatus> for enums::AttemptStatus {
+    fn from(item: BraintreePaymentStatus) -> Self {
+        match item {
+            BraintreePaymentStatus::Settling | BraintreePaymentStatus::Settled => Self::Charged,
+            BraintreePaymentStatus::AuthorizedExpired => Self::AuthorizationFailed,
+            BraintreePaymentStatus::Failed
+            | BraintreePaymentStatus::GatewayRejected
+            | BraintreePaymentStatus::ProcessorDeclined
+            | BraintreePaymentStatus::SettlementDeclined => Self::Failure,
+            BraintreePaymentStatus::Authorized => Self::Authorized,
+            BraintreePaymentStatus::Voided => Self::Voided,
+            _ => Self::Pending,
+        }
+    }
+}
+
+impl<F, T>
+    TryFrom<types::ResponseRouterData<F, BraintreePaymentsResponse, T, types::PaymentsResponseData>>
+    for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<
+            F,
+            BraintreePaymentsResponse,
+            T,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        if item.response.errors.is_some() {
```

Comment:
Instead of having errors and success response in single struct let's use enum over struct

Reply:
Created different structures to handle response